src/widget/woverview.cpp

```diff
@@ -93,9 +93,9 @@ void WOverview::setup(const QDomNode& node, const SkinContext& context) {
     m_marks.setup(m_group, node, context, m_signalColors);
 
     for (int i = 0; i < m_marks.size(); ++i) {
-        WaveformMark& mark = m_marks[i];
-        if (mark.m_pPointCos) {
-            mark.m_pPointCos->connectValueChanged(this,
+        WaveformMark* mark = m_marks[i].data();
```

Comment:
Or use _auto_:
auto pMark = m_marks[i];
This will create a temporary copy of the QSharedPointer and modifies the reference count, but that shouldn't matter. It's much more readable than anything else. I'm not sure if @daschuer agrees that _auto_ is appropriate here? We are inside the context of the same class and definitely (should) know what kind of elements we are storing in m_marks.